operator/src/main/java/oracle/kubernetes/operator/rest/resource/AdmissionWebhookResource.java

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator.rest.resource;
+
+import java.util.Optional;
+
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.MediaType;
+import oracle.kubernetes.operator.logging.LoggingFacade;
+import oracle.kubernetes.operator.logging.LoggingFactory;
+import oracle.kubernetes.operator.rest.model.AdmissionRequest;
+import oracle.kubernetes.operator.rest.model.AdmissionResponse;
+import oracle.kubernetes.operator.rest.model.AdmissionReview;
+import oracle.kubernetes.operator.rest.model.Status;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+import static oracle.kubernetes.common.logging.MessageKeys.VALIDATION_FAILED;
+import static oracle.kubernetes.operator.utils.GsonBuilderUtils.readAdmissionReview;
+import static oracle.kubernetes.operator.utils.GsonBuilderUtils.writeAdmissionReview;
+
+/**
+ * AdmissionWebhookResource is a jaxrs resource that implements the REST api for the /admission
+ * path. It is used as an endpoint for admission webhook, the API server will invoke
+ * this endpoint to validate a change request to a domain resource or cluster resource.
+ */
+@Path("admission")
+public class AdmissionWebhookResource extends BaseResource {
+
+  private static final LoggingFacade LOGGER = LoggingFactory.getLogger("Webhook", "Operator");
+
+  /** Construct a AdmissionWebhookResource. */
+  public AdmissionWebhookResource() {
+    super(null, "admission");
+  }
+
+  /**
+   * Validate a change request to a domain or cluster resource.
+   *
+   * @param body - a String representation of JSON document describing the AdmissionReview with admission request.
+   *
+   * @return a String representation of JSON document describing the AdmissionReview with admission response.
+   *
+   */
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  public String post(String body) {
```

Comment:
Why aren't AdmissionReview and AdmissionResponse the parameter and return type, respectively? (Why are you using String?)

Reply:
Similar to the conversion webhook request/response, we have our own model classes for the validating webhook. Those classes do not have all the fields that the K8S supports. We take the incoming request as a string and construct it to the internal request object and return the response as a String representing the response object. ConversationWebhookResource class does the same. 

I tried to use AdmissionReview and AdmissionResponse directly and got 400 (bad request) error.

Reply:
I can explore more and update the code in a future PR if we find a better solution.

Reply:
Do we really need the `entering` and `exiting` logs? I don't think we have ever found them useful.